src/plugins/intel_gpu/src/kernel_selector/cl_kernels/sdpa_opt.cl

```diff
@@ -224,7 +224,7 @@ KERNEL(sdpa_opt)(
                 const uint query_pitch = QUERY_STEP_LOCAL;
 #endif
 #if SG_SCALE_FACTOR == 2
-                if (sgid < HEAD_SIZE / SUBGROUP_SIZE) {
+                if (sgid < V_HEAD_SIZE / SUBGROUP_SIZE) {
 #else
                 {
 #endif
```

Comment:
What if v_head_size is smaller than k_head_size? Then we don't have enough subgroups to load all the required k_head_size query inputs